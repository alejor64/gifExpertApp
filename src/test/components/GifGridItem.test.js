import React from 'react'
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import GifGridItem from '../../components/GifGridItem'

describe('Purebas para el componente <GifGridItem />', () => {

    const title = 'Un titulo'
    const url = 'https://localhosto/img.png'
    const wrapper = shallow(<GifGridItem title={title} url={url}/>)
    
    test('Debe renderizar el componente <GifGridItem />', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe tener un parrafo con el Title', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    })
    
    test('Debe tener la images igual al url y alt de los props', () => {
        const img = wrapper.find('img')
        // console.log(img.props())

        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })
    
    test('Debe tener animate__fadeIn como clase', () => {
        const div = wrapper.find('div')
        const claseCSS = div.prop('className')
        
        expect(claseCSS.includes('animate__fadeIn')).toBe(true)
    })
    
})
