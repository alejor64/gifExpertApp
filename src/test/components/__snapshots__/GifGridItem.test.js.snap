// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Purebas para el componente <GifGridItem /> Debe renderizar el componente <GifGridItem /> 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhosto/img.png"
  />
  <p>
    Un titulo
  </p>
</div>
`;
